p2p/switch.go

```diff
@@ -529,13 +530,18 @@ func (sw *Switch) acceptRoutine() {
 				"max", sw.config.MaxNumInboundPeers,
 			)
 
-			_ = p.Stop()
+			_ = p.CloseConn()
+			sw.transport.Cleanup(p.RemoteAddr())
```

Comment:
Why not just pass the peer to Cleanup and let it call CloseConn and do the rest?